### PR TITLE
Remove Linux path names in ROCm search paths on Windows

### PR DESCRIPTION
**Pull request by david-salinas**
When target triple indicates we are on Windows, do not add linux
      paths to search path in ROCm Installation Detection.

Change-Id: I18effb8c20252de3d84ea37ef562124695c5a570

### CODE REVIEW

**Comment by yxsamliu** [comment]
better add a lit test like https://github.com/llvm/llvm-project/blob/main/clang/test/Driver/rocm-detect.hip , but for windows only (`REQUIRES: system-windows`), using `--print-rocm-search-dirs`, and checks `ROCm installation search path`.

**Comment by jayfoad** [comment]
> Compiler messages on HIP SDK for Windows

Please rewrite this to say what the patch does or what problem it fixes.

File: clang/test/Driver/rocm-detect-windows.hip [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// UNSUPPORTED: system-linux
```
[the remark this answers is not in the document]
Comment (david-salinas) [reply clang/test/Driver/rocm-detect-windows.hip]: changed to REQUIRES: system-windows in latest patch

**Comment by david-salinas** [comment]
> > Compiler messages on HIP SDK for Windows
> 
> Please rewrite this to say what the patch does or what problem it fixes.

Changed the PR (and commit) subject to something more meaningful/accurate.

**Review by yxsamliu — APPROVED** [review]

**Pull request closed without merging**